src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -106,6 +122,14 @@ def register_options(cls, register):
                 "`python_distribution`'s `entry_points` field."
             ),
         )
+        register(
+            "--unowned-dependency-behavior",
+            type=UnownedDependencyUsage,
+            default=UnownedDependencyUsage.DoNothing,
```

Comment:
This is a good+safe initial default, but I'm very interested in how many false positives you get with this... if we're able to make it warn by default in a future release, that would be great.

Reply:
Agreed! Will report back later on our monorepo

Reply:
So far it's just found a bunch of questionable imports, things that belong in `module_mapping`, and imports I hadn't realized we had that'll need some finagling.

An example of the last one was a dependency on an in-house compiled C extension module. Or funny import tricks like:

```
try:
    import foo
except ImportError:
    import bar
```

Reply:
Just finished with the monorepo. Nothing too surprising. The only one that got me was an import like `foo.bar.baz` being unowned. I added a `BUILD` file to `./foo/bar` (no `BUILD` files any higher) and declared a `python_requirement` with module `"baz"` (just to silence the warning, wouldn't pass the test) and still saw the error.

Instead I had to declare the module as `"foo.bar.baz"`. Not sure if that change makes sense, or works around Pants' shortcomings in some way :thinking:

Reply:
> I added a `BUILD` file to `./foo/bar` (no `BUILD` files any higher) and declared a `python_requirement` with module `"baz"` (just to silence the warning, wouldn't pass the test) and still saw the error.

I think that this makes sense... `python_requirement` targets do not have [source roots](https://www.pantsbuild.org/docs/source-roots) applied, so `foo.bar` would not have been prefixed onto `baz` in this case.